lstchain/tools/lstchain_create_dl3_file.py

```diff
@@ -0,0 +1,184 @@
+
+The selection cuts can be taken either from command-line arguments
+or a config file.
+
+Change the selection parameters as need be using the aliases.
+The default values are written in the DataSelection Component and
+in lstchain/data/data_selection_cuts.json
+"""
+
+from astropy.io import fits
+from astropy.coordinates import SkyCoord
+
+from ctapipe.core import Tool, traits, Provenance, ToolConfigurationError
+from lstchain.io import read_data_dl2_to_QTable
+from lstchain.reco.utils import get_effective_time
+from lstchain.paths import run_info_from_filename, dl2_to_dl3_filename
+from lstchain.irf import create_event_list
+from lstchain.io import EventSelector, DL3FixedCuts
+
+__all__ = ["DataReductionFITSWriter"]
+
+
+class DataReductionFITSWriter(Tool):
+    name = "DataReductionFITSWriter"
+    description = __doc__
+    example = """
+    To generate DL3 file from an observed data DL2 file, using default cuts:
+    > lstchain_create_dl3_file
+        -d /path/to/DL2_data_file.h5
+        -o /path/to/DL3/file/
+        -irf /path/to/irf.fits.gz
+        --source-name Crab
+        --source-ra 83.633deg
+        --source-dec 22.01deg
+
+    Or use a config file for the cuts:
+    > lstchain_create_dl3_file
+        -d /path/to/DL2_data_file.h5
+        -o /path/to/DL3/file/
+        -irf /path/to/irf.fits.gz
+        --source-name Crab
+        --source-ra 83.633deg
+        --source-dec 22.01deg
+        --overwrite
+        --config /path/to/config.json
+
+    Or pass the selection cuts from command-line:
+    > lstchain_create_dl3_file
+        -d /path/to/DL2_data_file.h5
+        -o /path/to/DL3/file/
+        -irf /path/to/irf.fits.gz
+        --source-name Crab
+        --source-ra 83.633deg
+        --source-dec 22.01deg
+        --gh 0.9
+        --theta 0.2
+        --overwrite
+    """
+
+    input_dl2 = traits.Path(
+        help="Input data DL2 file", exists=True, directory_ok=False, file_ok=True
+    ).tag(config=True)
+
+    output_dl3_path = traits.Path(
+        help="DL3 output filedir", directory_ok=True, file_ok=False
+    ).tag(config=True)
+
+    input_irf = traits.Path(
+        help="Compressed FITS file of IRFs",
+        exists=True,
+        directory_ok=False,
+        file_ok=True,
+    ).tag(config=True)
+
+    source_name = traits.Unicode(help="Name of Source").tag(config=True)
+
+    source_ra = traits.Unicode(help="RA position of the source").tag(config=True)
+
+    source_dec = traits.Unicode(help="DEC position of the source").tag(config=True)
+
+    overwrite = traits.Bool(
+        help="If True, overwrites existing output file without asking",
+        default_value=False,
+    ).tag(config=True)
+
+    classes = [EventSelector, DL3FixedCuts]
+
+    aliases = {
+        ("d", "input-dl2"): "DataReductionFITSWriter.input_dl2",
+        ("o", "output-dl3-path"): "DataReductionFITSWriter.output_dl3_path",
+        ("irf", "input-irf"): "DataReductionFITSWriter.input_irf",
```

Comment:
Short forms should be one letter. It is ok to not provide a short form.

Reply:
Ok. Amended.